base_layer/core/src/base_node/states/block_sync.rs

```diff
@@ -263,7 +306,13 @@ async fn synchronize_blocks<B: BlockchainBackend + 'static>(
 
             info!(target: LOG_TARGET, "Synchronize missing blocks.");
             let mut height = sync_height;
+            if let StateInfo::BlockSync(ref mut info) = shared.info {
+                *info.tip_height = Some(network_tip_height);
+            }
             while height <= network_tip_height {
+                if let StateInfo::BlockSync(ref mut info) = shared.info {
+                    *info.local_height = Some(height);
+                }
```

Comment:
Now when something interesting happens, we can push a message down the channel:

```rust
let event = StatusInfo::BlockSync(BlockSyncInfo::new(height, network_tip,...);
shared.status_event_subscriber(event).await;  // Awaiting might cause a bottleneck? But PoC for this review 
```

Reply:
done